File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/AI_finance/regional_treasury_scanner.py
class RegionalTreasuryScanner:
    """
    Analyzes if an AI country can afford regional membership fees.
    Regional organizations often have different budget impact priorities.
    """
    
    def __init__(self, threshold=0.15):
        self.threshold = threshold # 15% of treasury as max sustainable impact

    def scan_affordability(self, current_treasury, membership_fee):
        """
        Returns a detailed report on affordability.
        """
        if current_treasury <= 0:
            return {
                "can_afford": False,
                "is_sustainable": False,
                "impact_ratio": 1.0,
                "remaining_treasury": current_treasury,
                "reason": "Treasury empty."
            }

        impact_ratio = membership_fee / current_treasury
        can_afford = current_treasury >= membership_fee
        is_sustainable = impact_ratio <= self.threshold

        return {
            "can_afford": can_afford,
            "is_sustainable": is_sustainable,
            "impact_ratio": round(impact_ratio, 4),
            "remaining_treasury": current_treasury - membership_fee
        }

    def get_financial_recommendation(self, current_treasury, membership_fee, priority="normal"):
        """
        Provides a decision recommendation for the AI.
        """
        report = self.scan_affordability(current_treasury, membership_fee)
        
        if not report["can_afford"]:
            return "REJECT: Insufficient funds."

        ratio = report["impact_ratio"]

        # Regional organizations are often seen as more 'vital' for local stability
        if priority == "critical":
            if ratio < 0.40: return "APPROVE: Critical regional necessity."
            return "REJECT:suicidal cost."

        if ratio < 0.03: return "APPROVE: Negligible cost."
        if ratio < 0.10: return "APPROVE: Sustainable investment."
        if ratio < 0.25: return "HOLD: High impact for regional block."
        
        return "REJECT: Financial risk too high."

File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/AI_finance/RegionalTreasuryScanner.ts
/**
 * RegionalTreasuryScanner TS Bridge
 */
export interface RegionalTreasuryAnalysis {
    can_afford: boolean;
    is_sustainable: boolean;
    impact_ratio: number;
    remaining_treasury: number;
    reason?: string;
}

export class RegionalTreasuryScanner {
    private threshold: number;

    constructor(threshold = 0.15) {
        this.threshold = threshold;
    }

    public scanAffordability(currentTreasury: number, membershipFee: number): RegionalTreasuryAnalysis {
        if (currentTreasury <= 0) {
            return {
                can_afford: false,
                is_sustainable: false,
                impact_ratio: 1.0,
                remaining_treasury: currentTreasury,
                reason: "Treasury empty."
            };
        }

        const impactRatio = membershipFee / currentTreasury;
        const canAfford = currentTreasury >= membershipFee;
        const isSustainable = impactRatio <= this.threshold;

        return {
            can_afford: canAfford,
            is_sustainable: isSustainable,
            impact_ratio: Number(impactRatio.toFixed(4)),
            remaining_treasury: currentTreasury - membershipFee
        };
    }

    public getFinancialRecommendation(currentTreasury: number, membershipFee: number, priority: 'low' | 'normal' | 'high' | 'critical' = 'normal'): string {
        const report = this.scanAffordability(currentTreasury, membershipFee);
        if (!report.can_afford) return "REJECT: Insufficient funds.";

        const ratio = report.impact_ratio;

        if (priority === 'critical') {
            return ratio < 0.40 ? "APPROVE: Critical regional necessity." : "REJECT: suicidal cost.";
        }

        if (ratio < 0.03) return "APPROVE: Negligible cost.";
        if (ratio < 0.10) return "APPROVE: Sustainable investment.";
        if (ratio < 0.25) return "HOLD: High impact for regional block.";
        
        return "REJECT: Financial risk too high.";
    }
}
